training_record/serializers.py

```diff
@@ -129,6 +130,20 @@ def update(self, instance, validated_data):
         return RecordService.update_off_campus_record_from_raw_data(
             instance, **validated_data)
 
+    def validate(self, data):
+        '''Ensure having rights to update records'''
+        if not self.instance:
+            return data
+        user = self.context['request'].user
+        if ((user.is_teacher and
+             is_user_allowed_operating(self.context['request'],
+                                       self.instance)) or
+                ((user.is_department_admin or user.is_school_admin) and
+                 is_admin_allowed_operating(self.context['request'],
+                                            self.instance))):
+            return data
+        raise BadRequest('在此状态下您无法更改。')
```

Comment:
raise ValidationError.
No need to check `user.is_teacher` or `user.is_department_admin or user.is_school_admin`, they have already been checked in these functions.